subscription-sender/internal/app/app.go

```diff
@@ -24,19 +29,38 @@ func NewApp(cfg Config, logger tools.Logger, ctx context.Context) (*App, error)
 
 	subscriptionService := services.NewSubscriptionService(logger, emailSender)
 
-	notifyHandler := handlers.NewNotifyHandler(logger, subscriptionService)
+	healthCheckHandler := handlers.NewHealthCheckHandler(logger)
+
+	router := routers.NewRouter(logger, healthCheckHandler)
+
+	nats, err := nats.Connect(cfg.NatsURL)
+	if err != nil {
+		return nil, err
+	}
 
-	router := routers.NewRouter(logger, notifyHandler)
+	queue := queue.NewQueue(nats, cfg.SubscriptionTopicName, subscriptionService, logger)
 
 	return &App{
 		cfg: cfg,
 		logger: logger,
 		router: router,
+		queue: queue,
+		ctx: ctx,
 	}, nil
 }
 
 func (app *App) Run() error {
 	app.logger.Info("app run")
+	
+	defer func() { 
+		_ = app.queue.QueueConn.Drain()
+	}()
 
+	err := app.queue.ConsumeSubscriptionEvent(app.ctx)
+
+	if err != nil {
+        app.logger.Error("Queue subscribe method faild")
```

Comment:
seems like not ok if you application is failed to run and you just log it.

Reply:
Done